projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/handlers.py

```diff
@@ -38,14 +37,9 @@ def post(self):
 class DownloadJobHandler(APIHandler):
     @tornado.web.authenticated
     def post(self):
-        input_data = self.get_json_body()
+        job_data = DictObj(self.get_json_body())
         try:
-            status = VdkUI.download_job(
-                input_data["jobName"],
-                input_data["jobTeam"],
-                input_data["restApiUrl"],
-                input_data["parentPath"],
-            )
+            status = VdkUI.download_job(job_data)
```

Comment:
I fancy the (first) approach of having the needed-only exact parameters, rather a blob of a generic object;

Reply:
Thinking about input_data[JOB_NAME], where "jobName" is not a free-text value -such is prone to errors

Reply:
If we're enforcing a specific object model, we shouldn't be worried about passing the correct data to methods like this one. This is a lot better imo, instead of having a method signature with 5 parameters, some of which are optional and have default values.

Reply:
Three possibilities to design interface surface for those methods `VdkUI.*` (for example `VdkUI.download_job()`): 
1. use simple data typed params (like before, jobName: str, ...)
2. use a class type, per method; for example, DownloadJobDto (that itself is composite of the params needed)
3. use a generic object that could/could not contain the params needed -> and assert on the params requires do exist during method implementation (the way latest change attempts to do, which I am not in favour of, due poor data contract)

I am looking to have the most simplistic, per key access of `input_data[KEY]`, where KEY is a constant (could be easily reused, and is subjected to compiler changes, and could be accompanied with a description).